### PR TITLE
Adaptive Demo [Material 3 Reply App]

### PR DESCRIPTION
**Pull request by a-wallen**
Out-of-the-box adaptive layouts were the most requested feature on desktop platforms in the Q3 2022 survey. The M3 Reply App is a good example of an app that adapts. This PR contains an implementation of the app.

https://user-images.githubusercontent.com/44445638/206802465-d38703be-13bd-42b4-b041-2e6d52436e10.mp4

https://user-images.githubusercontent.com/44445638/206802524-55d3d1c3-b34c-4aa8-ab77-8ae8dfe18724.mov

## Pre-launch Checklist

- [ ] I read the [Effective Dart: Style] _recently_, and have followed its advice.
- [X] I signed the [CLA].
- [X] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

[Effective Dart: Style]: https://dart.dev/guides/language/effective-dart/style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: material3-reply/lib/models/name.dart [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Name {
+  const Name({
+    required this.first,
+    required this.last,
+  });
+
+  final String first;
+  final String last;
+  String get fullName => "$first $last";
+  String get initials => throw UnimplementedError();
```
Comment (domesticmouse) [inline material3-reply/lib/models/name.dart:10]: drop?

File: material3-reply/lib/animation/constants.dart [hunk 2]
```diff
@@ -0,0 +1,27 @@
+enum MaterialDuration {
```
Comment (domesticmouse) [inline material3-reply/lib/animation/constants.dart:1]: Please add a license header to this file and the other Dart files, per https://github.com/flutter/codelabs/blob/main/CONTRIBUTING.md#file-headers

Yes, I know the year is wrong, please see https://github.com/flutter/codelabs/pull/1208 for the fix =)

File: material3-reply/test/widget_test.dart [hunk 3]
```diff
@@ -0,0 +1,30 @@
+// This is a basic Flutter widget test.
+//
+// To perform an interaction with a widget in your test, use the WidgetTester
+// utility in the flutter_test package. For example, you can send tap and scroll
+// gestures. You can also use WidgetTester to find child widgets in the widget
+// tree, read text, and verify that the values of widget properties are correct.
+
+import 'package:flutter/material.dart';
```
Comment (domesticmouse) [inline material3-reply/test/widget_test.dart:8]: I suspect this wont pass CI. 

Please add this project to https://github.com/flutter/codelabs/blob/main/flutter_ci_script_beta.sh and add a commented out line to https://github.com/flutter/codelabs/blob/main/flutter_ci_script_stable.sh

**Review by domesticmouse — CHANGES_REQUESTED** [review]

**Comment by domesticmouse** [comment]
CI Issues:

```console
Analyzing material3-reply...

   info - lib/widgets/email_widget.dart:103:65 - The 'child' argument should be last in widget constructor invocations. Try moving the argument to the end of the argument list. - sort_child_properties_last
   info - lib/widgets/email_widget.dart:103:72 - Use 'const' with the constructor to improve performance. Try adding the 'const' keyword to the constructor invocation. - prefer_const_constructors
   info - lib/widgets/email_widget.dart:109:65 - The 'child' argument should be last in widget constructor invocations. Try moving the argument to the end of the argument list. - sort_child_properties_last
   info - lib/widgets/email_widget.dart:109:72 - Use 'const' with the constructor to improve performance. Try adding the 'const' keyword to the constructor invocation. - prefer_const_constructors

4 issues found.
```

**Comment by a-wallen** [comment]
@domesticmouse thanks for the ping. I think that https://github.com/flutter/codelabs/pull/1214/commits/56e5fd5b24e5e97bfc386860fd5a152495b09693 should fix those analyzer errors.

**Comment by domesticmouse** [comment]
And yet more errors. Please run `dart format` on the code, it's enforced by the CI.

**Review by domesticmouse — APPROVED** [review]

**Pull request merged**